zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -177,6 +176,16 @@ public InterpreterSettingManager(ZeppelinConfiguration conf,
     this.interpreterEventServer = new RemoteInterpreterEventServer(conf, this);
     this.interpreterEventServer.start();
     init();
+
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread(
```

Comment:
so then this would run asynchronously (because new thread) in the shutdown hook? doesn't that mean it is possible that it hasn't shutdown completely then?

Reply:
don't we need to call .start() on the new thread?

Reply:
Yes, we don't have to do.